aten/src/ATen/cuda/CUDAEvent.h

```diff
@@ -45,6 +46,10 @@ struct TORCH_CUDA_CPP_API CUDAEvent {
     try {
       if (is_created_) {
         CUDAGuard guard(device_index_);
+        const auto* interp = c10::impl::CUDATraceTLS::get_trace();
+        if (interp) {
```

Comment:
Nit: I believe there exist some `C10_LIKELY` and `C10_UNLIKELY` macros that can be used to tag some conditions that we believe have a tendency to usually be true/false, in order to "nudge" the compiler and the branch prediction to optimize for the most common case.

They tend also to be used in cases like this when we want a no-op fast-path be the least impacted by a change. Hence you could consider using `C10_UNLIKELY` in these cases.

Reply:
Thank you for mentioning that, I will use them.